EnergyPlus_Engine/Query/ReadESO.cs

```diff
@@ -0,0 +1,107 @@
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace BH.Engine.EnergyPlus
+{
+    public static partial class Query
+    {
+        [Description("Read an EnergyPlus simulation results ESO file")]
+        [Input("filepath", "The path to the ESO file")]
+        [Output("esoFile", "The ESO file object containing simulation results")]
+        public static ESOFile ReadESO(string filepath)
+        {
+            ESOFile esoFile = new ESOFile();
+            esoFile.FilePath = filepath;
+
+            // Load ESO file into list for parsing
+            List<string> lines = File.ReadAllLines(filepath, Encoding.UTF8).ToList();
+
+            // Parse header for metadata
+            string version = lines[0].Split(',')[2].Trim().Replace("Version ", "");
+            esoFile.EnergyPlusVersion = version;
+            string creationDate = lines[0].Split(',')[3].Trim().Replace("YMD=", "");
+            esoFile.CreationDate = DateTime.Parse(creationDate);
+
+            // Obtain results data dictionary
+            int ddStartIndex = 7;
+            int ddEndIndex = 1;
+            for (int i = 0; i < lines.Count; i++)
+            {
+                if (lines[i] == "End of Data Dictionary")
+                {
+                    ddEndIndex = i;
+                }
+            }
+            List<string> dataDictionary = lines.GetRange(ddStartIndex, ddEndIndex - ddStartIndex);
+
+            // Obtain results keys
+            List<int> resultsIndices = new List<int>();
+            List<string> resultsKeys = new List<string>();
+            List<string> resultsAtrributes = new List<string>();
+            foreach (string i in dataDictionary)
+            {
+                string[] kvPair = i.Split(',');
+                resultsIndices.Add(Int32.Parse(kvPair[0].Trim()));
+                resultsKeys.Add(kvPair[2].Trim());
+                resultsAtrributes.Add(kvPair[3]);
+            }
+
+            // Obtain results value lists
+            List<string> valueDictionary = lines.GetRange(ddEndIndex + 1, lines.Count() - 3 - ddEndIndex);
+            Dictionary<int, List<double>> resultsValues = new Dictionary<int, List<double>>();
+            foreach (int idx in resultsIndices)
+            {
+                resultsValues.Add(idx, new List<double>());
+            }
+
+            foreach (string i in valueDictionary)
+            {
+                string[] kvPair = i.Split(',');
+                int key = Int32.Parse(kvPair[0]);
+
+                if (!resultsIndices.Contains(key))
+                {
+                    continue;
+                }
+
+                double value = Double.Parse(kvPair[1]);
+                resultsValues[key].Add(value);
+                
+            }
+
+            esoFile.ResultsAttribute = resultsAtrributes;
+            esoFile.ResultsObjects = resultsKeys;
+            esoFile.ResultsValues = resultsValues.Values.ToList();
```

Comment:
I suspect this should probably be done by the adapter, but happy to leave it as is for now to kick start development and we can look at the right method next week